File: app/schemas/rejection.py
import uuid
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class RejectionOut(BaseModel):
    id: uuid.UUID
    client_id: uuid.UUID
    client_name: Optional[str] = None
    content_id: uuid.UUID
    content_name: Optional[str] = None
    content_type: Optional[str] = None
    reason: str
    rejected_by: Optional[str] = None
    rejected_at: datetime
    resolved_at: Optional[datetime] = None
    resolution_status: str
    notes: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class RejectionResolveRequest(BaseModel):
    resolution_notes: Optional[str] = None

File: frontend/src/types/rejection.ts
// ============================================================
// Company OS — Rejection Types
// ============================================================

export interface Rejection {
  id: string;
  client_id: string;
  client_name?: string;
  content_id: string;
  content_name?: string;
  content_type?: string;
  reason: string;
  rejected_by?: string;
  rejected_at: string;
  resolved_at?: string;
  resolution_status: string;
  notes?: string;
}

export interface RejectionResolveRequest {
  resolution_notes?: string;
}
